src/net.cpp

```diff
@@ -3027,6 +3028,9 @@ CNode::~CNode()
     if (nMisbehavior == 0 && successfullyConnected())
         addrman.Connected(addr);
 
+    // Decrement thintype peer counters
+    thinrelay.RemovePeers(this);
+
```

Comment:
In this PR , this is the significant line which maybe/probably solves the occasional crash on exit we've been getting reports on. We have to do this here in the destructor rather than in GetNodeSignals.FinalizeNode() which is called in the next line.